hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+        {
+            Port<"port-3">: [("vlan/native", "23"), ("vlan/52", "52")],
+            Port<"port-7">: [("vlan/23", "23")],
+            Port<"port-8">: [("vlan/native", "52")],
+            ...
+        }
+
+        """
+        response = {}
+        for port in ports:
+            response[port] = filter(None,
+                                    [self._get_native_vlan(port.label)]) \
+                                    + self._get_vlans(port.label)
+        return response
+
+    def _get_vlans(self, interface):
+        """ Return the vlans of a trunk port.
+
+        Does not include the native vlan. Use _get_native_vlan.
+
+        Args:
+            interface: interface to return the vlans of
+
+        Returns: List containing the vlans of the form:
+        [('vlan/vlan1', vlan1), ('vlan/vlan2', vlan2)]
+
+        It uses the REST API CLI which is slow but it is the only way to do it
+        because the switch is VLAN centric. Doing a GET on interface won't
+        return the VLANS on it, we would have to do get on all vlans (if that
+        worked reliably in the first place) and then find our interface there
+        which is not feasible.
+        """
+
+        url = self._construct_url()
+        command = 'interfaces switchport %s %s' % \
+            (self.interface_type, interface)
+        payload = self._make_payload(SHOW, command)
+        response = self._make_request('POST', url, data=payload)
+        # parse the output to get a list of all vlans.
+        response = response.text.replace(' ', '').splitlines()
+        # should probably make this more reliable
+        try:
+            index = response.index("Vlanmembership:") + 3
+        except ValueError:
+            return []
+        if response[index] == '':
+            # a port with no tagged interface has no section with "T",
+            # it's just an empty line.
+            return []
+        vlan_list = response[index].replace('T', '').split(',')
+        return [('vlan/%s' % x, x) for x in vlan_list]
+
+    def _get_native_vlan(self, interface):
+        """ Return the native vlan of an interface.
+
+        Args:
+            interface: interface to return the native vlan of
+
+        Returns: Tuple of the form ('vlan/native', vlan) or None
+
+        Similar to _get_vlans()
+        """
+
+        url = self._construct_url()
+        command = 'interfaces switchport %s %s' % \
+            (self.interface_type, interface)
+        payload = self._make_payload(SHOW, command)
+        response = self._make_request('POST', url, data=payload)
+        # find the Native Vlan ID from the response
+        response = response.text.replace(' ', '')
+        if response.find('NativeVlanId:') == -1:
+            return None
+        begin = response.find('NativeVlanId:') + len('NativeVlanId:')
+        end = response.find('.', begin)
+        vlan = response[begin:end]
+        # a really bad temporary workaround to get the deployment tests to pass
```

Comment:
Explain what it's doing, why exactly the tests are failing without it, and what's wrong with it.

Reply:
Do you want me to add it as a comment here or explain it to you (because I am pretty sure you already know what's up) ?

Reply:
as a comment, sorry if that wasn't clear.